pkg/devfile/versions/1.0.1-beta/types.go

```diff
@@ -0,0 +1,230 @@
+
+	// Additional command attributes
+	Attributes Attributes `yaml:"attributes,omitempty" json:"attributes,omitempty"`
+
+	// Describes the name of the command. Should be unique per commands set.
+	Name string `yaml:"name"`
+
+	// Preview url
+	PreviewUrl DevfileCommandPreviewUrl `yaml:"previewUrl,omitempty" json:"previewUrl,omitempty"`
+}
+
+type DevfileCommandPreviewUrl struct {
+	Port *string `yaml:"port,omitempty" json:"port,omitempty"`
+	Path *string `yaml:"path,omitempty" json:"path,omitempty"`
+}
+
+type DevfileCommandAction struct {
+
+	// The actual action command-line string
+	Command *string `yaml:"command,omitempty" json:"command,omitempty"`
+
+	// Describes component to which given action relates
+	Component *string `yaml:"component,omitempty" json:"component,omitempty"`
+
+	// the path relative to the location of the devfile to the configuration file
+	// defining one or more actions in the editor-specific format
+	Reference *string `yaml:"reference,omitempty" json:"reference,omitempty"`
+
+	// The content of the referenced configuration file that defines one or more
+	// actions in the editor-specific format
+	ReferenceContent *string `yaml:"referenceContent,omitempty" json:"referenceContent,omitempty"`
+
+	// Describes action type
+	Type *DevfileCommandType `yaml:"type,omitempty" json:"type,omitempty"`
+
+	// Working directory where the command should be executed
+	Workdir *string `yaml:"workdir,omitempty" json:"workdir,omitempty"`
+}
+
+type DevfileComponent struct {
+
+	// The name using which other places of this devfile (like commands) can refer to
+	// this component. This attribute is optional but must be unique in the devfile if
+	// specified.
+	Alias *string `yaml:"alias,omitempty" json:"alias,omitempty"`
+
+	// Describes whether projects sources should be mount to the component.
+	// `CHE_PROJECTS_ROOT` environment variable should contains a path where projects
+	// sources are mount
+	MountSources bool `yaml:"mountSources,omitempty" json:"mountSources,omitempty"`
+
+	// Describes type of the component, e.g. whether it is an plugin or editor or
+	// other type
+	Type DevfileComponentsType `yaml:"type" json:"type"`
+
+	// for type ChePlugin
+	DevfileComponentChePlugin `yaml:",inline" json:",inline"`
+
+	// for type=dockerfile
+	DevfileComponentDockerimage `yaml:",inline" json:",inline"`
+}
+
+type DevfileComponentChePlugin struct {
+	Id          *string `yaml:"id,omitempty" json:"id,omitempty"`
+	Reference   *string `yaml:"reference,omitempty" json:"reference,omitempty"`
+	RegistryUrl *string `yaml:"registryUrl,omitempty" json:"registryUrl,omitempty"`
+}
+
+type DevfileComponentCheEditor struct {
+	Id          *string `yaml:"id,omitempty" json:"id,omitempty"`
+	Reference   *string `yaml:"reference,omitempty" json:"reference,omitempty"`
+	RegistryUrl *string `yaml:"registryUrl,omitempty" json:"registryUrl,omitempty"`
+}
+
+type DevfileComponentOpenshift struct {
```

Comment:
Can we reuse all these strings from `1.0.0`, maybe even extract them into a common packge? and at places where the structs have more parameters can be embed the `1.0.0` struct in the respective `1.0.1` struct and add the new parameter in the `1.0.1` struct?

Reply:
This approach will help use use a common interface and not have branches in the layers of code which uses devfiles

Reply:
This might be actually a good idea. Not sure about a common package, but reusing structs from the previous version if they are not changes make actually sense. It will also nicely highlight the changes between versions.

Reply:
I have refactored the code to have common structs and common functions between different versions of devfile.

As per our discussion with the devfile team, since the `1.0.1-beta` version has been merged into `1.0.0` version, I have removed the `1.0.1-beta` files from the code as well.